_posts/2024-12-xx-hybrid-search-optimization.md

```diff
@@ -0,0 +1,342 @@
+
+We again use the l2 norm and arithmetic mean as our hybrid search normalization and combination parameter values as they scored best in the global hybrid search optimizer experiment. With that we build the hybrid query, execute it, retrieve the results and calculate the search metrics like in the baseline and global hybrid search optimizer. 
+
+Metrics for the small dataset:
+
+| Metric | Baseline BM25 | Global Hybrid Search Optimizer | Dynamic Hybrid Search Optimizer - Linear Model | Dynamic Hybrid Search Optimizer - Random Forest Model |
+| :---: | :---: | :---: | :---: | :---: |
+| DCG@10 | 9.65 | 9.99 | 10.92 | 10.92 |
+| NDCG@10 | 0.24 | 0.26 | 0.28 | 0.28 |
+| Precision@10 | 0.27 | 0.29 | 0.32 | 0.32 |
+
+Metrics for the large dataset:
+
+| Metric | Baseline BM25 | Global Hybrid Search Optimizer | Dynamic Hybrid Search Optimizer - Linear Model | Dynamic Hybrid Search Optimizer - Random Forest Model |
+| :---: | :---: | :---: | :---: | :---: |
+| DCG@10 | 8.82 | 9.30 | 10.13 | 10.13 |
+| NDCG@10 | 0.23 | 0.25 | 0.27 | 0.27 |
+| Precision@10 | 0.24 | 0.27 | 0.29 | 0.29 |
+
+Looking at these numbers shows us a steady positive trend starting from the baseline going all the way to the dynamic predictions of lexicalness and neuralness per query. The large dataset shows a DCG increase of 8.9% rising from 9.3 to 10.13, the small dataset shows an increase of 9.3%. The other metrics increase as well: NDCG shows an improvement of 7.4%for the large dataset, 10.3% for the small dataset, Precision shows an improvement of 8% for the large dataset and 7.7% for the small dataset.
+
+Interestingly, both models score exactly equally. The reason for this is that while they both predict different NDCG values, they predict the best ones with the same “neuralness” as an input feature. So while the models may differ in RMSE scores during the evaluation phase, they provide equal results when applied to the test set.
+
+Despite the low judgement coverage we see improvements for all metrics. This gives us confidence that this approach can provide value for search systems not only switching from lexical to hybrid search but also those who already are in production but have never used any systematic process to evaluate and identify the best settings.
+
+# Conclusion
+
+We provide a systematic approach to optimizing hybrid search in OpenSearch based on its current state and capabilities (normalization & combination techniques). The results look promising especially given the low judgment coverage that the ESCI dataset has.
+
+We encourage everyone to adopt the approach and explore its usefulness in their domain with their dataset. We are looking forward to hearing about the experimentation results the community has with the provided approach.
+
```

Comment:
Line 327: Should we include how they can provide their feedback (link)?

Reply:
There's no dedicated feedback or anything similar that I see appropriate to link here. Do you have a suggestion? Something like reach out to the author on OpenSearch Slack?

Reply:
Maybe on the [OpenSearch forum](https://forum.opensearch.org/)?